File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/LineageNode.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// LineageNode Declaration
//==============================================================
type LineageNode struct {
    gorm.Model
     Name                                    string
    QualifiedName                                    string
    WorkspaceId         *uint
    Workspace           *AnalyticsWorkspace `gorm:"foreignKey:WorkspaceId"`
     Inputs           []LineageNode `gorm:"foreignKey:InputsFromLineageNodeId"`
     Outputs           []LineageNode `gorm:"foreignKey:OutputsFromLineageNodeId"`
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromLineageNodeId"`
     Models           []Model_ `gorm:"foreignKey:ModelsFromLineageNodeId"`
     Pipelines           []DataPipeline `gorm:"foreignKey:PipelinesFromLineageNodeId"`
     Dashboards           []Dashboard `gorm:"foreignKey:DashboardsFromLineageNodeId"`
     Reports           []Report `gorm:"foreignKey:ReportsFromLineageNodeId"`
    NodeType                      LineageNodeType

// parent associations as their child

}

File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/LineageNode.js

// Define collection and schema for LineageNode
export interface LineageNode {
    name:
	type : string
    qualifiedName:
	type : string
    Workspace:
	type : Schema.Types.ObjectId
    Inputs:
 	type : [{ type: Schema.Types.ObjectId, ref: 'LineageNode' }]
    Outputs:
 	type : [{ type: Schema.Types.ObjectId, ref: 'LineageNode' }]
    Datasets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSet' }]
    Models:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Model_' }]
    Pipelines:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataPipeline' }]
    Dashboards:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Dashboard' }]
    Reports:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Report' }]
    NodeType:
 	type : String
#
    collection: 'lineageNodes'
}
